src/common/NetworkManager.ts

```diff
@@ -130,14 +161,21 @@ export class NetworkManager extends EventEmitter {
   }
 
   async setOfflineMode(value: boolean): Promise<void> {
-    if (this._offline === value) return;
-    this._offline = value;
+    await this.emulateNetworkConditions(
+      value ? networkConditionsMap['Offline'] : networkConditionsMap['Online']
```

Comment:
Calling `setOfflineMode(true); setOfflineMode(true);` removes conditions previously set by the command `emulateNetworkConditions`, which can be an unexpected side-effect.

Reply:
This is a good point! I'll change the two apis to not interfere with each other.